### ISSUE TITLE
Handover should not apply limits

### ISSUE THREAD

**Issue by gouthamve** [cortexproject/cortex#1349]
We found one of the rollouts failed with:
```
level=info ts=2019-04-25T15:42:47.885425968Z caller=lifecycler.go:456 msg="changing ingester state from" old_state=PENDING new_state=JOINING
level=info ts=2019-04-25T15:42:47.918991563Z caller=transfer.go:83 msg="processing TransferChunks request" from_ingester=ingester-6bcf4b69cf-wz6sp

level=error ts=2019-04-25T15:43:55.494427327Z caller=transfer.go:54 msg="TranferChunks failed, not in ACTIVE state." state=JOINING
level=info ts=2019-04-25T15:43:55.494516868Z caller=lifecycler.go:456 msg="changing ingester state from" old_state=JOINING new_state=PENDING
level=warn ts=2019-04-25T15:45:23.717665467Z caller=grpc_logging.go:47 method=/cortex.Ingester/TransferChunks duration=55.413853328s err="rpc error: code = Code(429) desc = per-user series limit (500000) exceeded" msg="gRPC\n"
```

We should not apply limits during handover. 

Further, the sending ingester should not have more series than the limits. We maybe hitting this "harmless" race here: https://github.com/cortexproject/cortex/blob/590e72c616def79c35c9add08ee3e9d28e1349f0/pkg/ingester/user_state.go#L190-L194

**Comment by bboreham** [comment]
I had a similar-looking issue, that turned out to be caused by the race fixed at #1703.

**Comment by alanprot** [comment]
Closing as this is related to chunks storage and its being deprecated.
